app/auth.go

```diff
@@ -82,19 +143,202 @@ func NewAuthManager(conf AuthManagerConfig) AuthManager {
 	return mgr
 }
 
-func (m *MenderAuthManager) IsAuthorized() bool {
-	adata, err := m.AuthToken()
+// GetInMessageChan returns the channel to send requests to the auth manager
+func (m *MenderAuthManager) GetInMessageChan() chan<- *AuthManagerRequest {
+	return m.inChan
+}
+
+// GetBroadcastMessageChan returns the channel to get responses from the auth manager
+func (m *MenderAuthManager) GetBroadcastMessageChan(name string) <-chan *AuthManagerResponse {
+	if m.broadcastChans[name] == nil {
+		m.broadcastChans[name] = make(chan *AuthManagerResponse, 1)
+	}
+	return m.broadcastChans[name]
+}
+
+// Run is the main routine of the Mender authorization manager
+func (m *MenderAuthManager) Run() error {
+	m.running = true
+	for m.running {
+		select {
```

Comment:
Actually, could not the channel send the struct, instead of a pointer to the struct, and benefit from the standard values? Then the ok check is not needed?

Reply:
Switch was a left-over from a previous implementation, with multiple channels, thanks for pointing it out!
I switched to struct instead of the pointer.